code/game/machinery/casino_ch.dm

```diff
@@ -0,0 +1,900 @@
+	if(W.is_wrench())
+		playsound(src, W.usesound, 50, 1)
+		user << "<span class='notice'>You begin to [anchored ? "un" : ""]fasten \the [src].</span>"
+		if (do_after(user, 20 * W.toolspeed))
+			user.visible_message(
+				"<span class='notice'>\The [user] [anchored ? "un" : ""]fastens \the [src].</span>",
+				"<span class='notice'>You have [anchored ? "un" : ""]fastened \the [src].</span>",
+				"You hear a ratchet.")
+			anchored = !anchored
+		else
+			user << "<span class='notice'>You decide not to [anchored ? "un" : ""]fasten \the [src].</span>"
+
+	else if(istype(W, /obj/item/weapon/reagent_containers/chem_disp_cartridge))
+		add_cartridge_deluxe(W, user)
+
+	else if(W.is_screwdriver())
+		var/label = input(user, "Which cartridge would you like to remove?", "Chemical Dispenser") as null|anything in cartridges
+		if(!label) return
+		var/obj/item/weapon/reagent_containers/chem_disp_cartridge/C = remove_cartridge(label)
+		if(C)
+			user << "<span class='notice'>You remove \the [C] from \the [src].</span>"
+			C.loc = loc
+			playsound(src, W.usesound, 50, 1)
+
+	else if(istype(W, /obj/item/weapon/reagent_containers/glass) || istype(W, /obj/item/weapon/reagent_containers/food))
+		if(container)
+			user << "<span class='warning'>There is already \a [container] on \the [src]!</span>"
+			return
+
+		var/obj/item/weapon/reagent_containers/RC = W
+
+		if(!accept_drinking && istype(RC,/obj/item/weapon/reagent_containers/food))
+			user << "<span class='warning'>This machine only accepts beakers!</span>"
+			return
+
+		if(!RC.is_open_container())
+			user << "<span class='warning'>You don't see how \the [src] could dispense reagents into \the [RC].</span>"
+			return
+
+		container =  RC
+		user.drop_from_inventory(RC)
+		RC.loc = src
+		user << "<span class='notice'>You set \the [RC] on \the [src].</span>"
+		SSnanoui.update_uis(src)
+
+	else
+		return ..()
+
+/obj/machinery/vending/deluxe_boozeomat
+	name = "Deluxe Drink Distributor"
+	desc = "A top of the line and experimental drink vendor, it uses bluespace technology for storage!"
+	icon = 'icons/obj/casino_ch.dmi'
+	icon_state = "deluxe_boozeomat"
+	icon_deny = "deluxe_boozeomat_deny"
+	products = list(/obj/item/weapon/reagent_containers/food/drinks/glass2/square = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/rocks = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/shake = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/cocktail = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/shot = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/pint = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/mug = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/wine = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/metaglass = 25,
+					/*/obj/item/weapon/reagent_containers/food/drinks/metaglass/metapint = 25,*/ //Metapint glass not ported yet - Jack
```

Comment:
Aaaaaaaand here as well.

Reply:
Done, moved the comments out to avoid issues.